blivet/util.py

```diff
@@ -116,12 +118,39 @@ def mount(device, mountpoint, fstype, options=None):
 
     return rc
 
-def umount(mountpoint):
-    try:
-        rc = run_program(["umount", mountpoint])
-    except OSError:
-        raise
+def umount(mountpoint, lazy=False, maxretry=3, retrysleep=1.0):
+    """ Unmount the given mountpoint.
 
+        :param str mountpoint: Path to unmount
+        :param bool lazy: When True do a lazy (-l) umount
+        :param int maxretry: Number of times to try the umount
+        :param float retrysleep: Seconds to sleep between retries
+        :returns: rc of last umount attempt
+        :rtype: int
+        :raises CalledProcessError: if all the retries fail
+        :raises OSError: if command cannot be executed
+    """
+    cmd = ["umount"]
+    if lazy: cmd += ["-l"]
+    cmd += [mountpoint]
+    count = 0
+    while maxretry > 0:
```

Comment:
What is this condition good for? If ``maxretry`` (or ``maxtry``) is ``0``, the function can just return early.